src/main/java/org/broadinstitute/hellbender/utils/downsampling/PositionalDownsampler.java

```diff
@@ -109,9 +128,11 @@ public GATKRead peekFinalized() {
 
     @Override
     public GATKRead peekPending() {
-        // The finalized items in the ReservoirDownsampler are pending items from the perspective of the
-        // enclosing PositionalDownsampler
-        return reservoir.peekFinalized();
+        // The ReservoirDownsampler accumulates pending items until signalEndOfInput has been called, at which
+        // point all items that have survived downsampling become finalized. From the perspective of the enclosing
+        // PositionalDownsampler, both finalized items and pending items in the ReservoirDownsampler are considered
+        // pending.
+        return Optional.ofNullable(reservoir.peekFinalized()).orElse(reservoir.peekPending());
     }
 
     @Override
```

Comment:
When `signalEndOfInput()` is called on a `PositionalDownsampler`, should it then forbid subsequent attempts to submit additional items, to mirror the new `ReservoirDownsampler` behavior?